### PR TITLE
Input: add show-word-count attribute

### PR DESCRIPTION
**Pull request by luckyCao**
Please make sure these boxes are checked before submitting your PR, thank you!

* [ ] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md) | [Español](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.fr-FR.md)).
* [ ] Make sure you are merging your commits to `dev` branch.
* [ ] Add some descriptions and refer relative issues for you PR.

### CODE REVIEW

**Comment by wacky6** [comment]
Prefer not to have word count in mainline.

The definition of a word may differ. In English one word consists of multiple characters. In Chinese, a word usually means a single glphy.

Also, JavaScript `.length` is the number of UTF-16 code units, it does not work for complex characters (surrogate pairs). (e.g. `'😀'.length === 2`, `'𥑮'.length === 2`). To get the number of code points (real character / glyph), use `[...string].length`.

---

`maxlength` is passed to native input and already limits `value.length`, so `isExceed` will never be true. Also, `maxlength` should be coerced into a number in `upperLimit`.

Shouldn't word count work without `maxlength` ? User may only want to hint character count rather than to enforce a strict limit. 

Does it conflict with `clearable` / other suffix slot usage?

**Comment by luckyCao** [comment]
Thanks remind complex characters, will fix it.
When initial value's length is larger then upperLimit, isExceed is true. 
It's not conflict with clearable / other suffix slot

File: packages/input/src/input.vue [hunk 1]
```diff
@@ -48,9 +49,9 @@
       <!-- 后置内容 -->
       <span
         class="el-input__suffix"
-        v-if="$slots.suffix || suffixIcon || showClear || showPassword || validateState && needStatusIcon">
+        v-if="$slots.suffix || suffixIcon || showClear || showPassword || showWordCountVisible || validateState && needStatusIcon ">
```
Comment (ziyoung) [inline packages/input/src/input.vue]: 优化一下。这块判断太复杂。

File: types/input.d.ts [hunk 2]
```diff
@@ -81,6 +81,15 @@ export declare class ElInput extends ElementUIComponent {
...
   validateEvent: boolean
 
+  /** Whether the input is clearable */
+  clearable?: boolean
+
+  /** Whether to show password */
+  showPassword?: boolean
+
+  /** Whether to show wordCount when setting maxLength */
+  showWordCount?: boolean
```
Comment (SimonaliaChen) [inline types/input.d.ts]: prefer showWordLimit

File: types/input.d.ts [hunk 3]
```diff
@@ -81,6 +81,15 @@ export declare class ElInput extends ElementUIComponent {
   /** Whether to trigger form validatio */
   validateEvent: boolean
 
+  /** Whether the input is clearable */
+  clearable?: boolean
```
Comment (SimonaliaChen) [inline types/input.d.ts]: is there any need to add question mark?

File: packages/input/src/input.vue [hunk 4]
```diff
@@ -49,9 +49,9 @@
       <!-- 后置内容 -->
       <span
         class="el-input__suffix"
-        v-if="$slots.suffix || suffixIcon || showClear || showPassword || showWordCountVisible || validateState && needStatusIcon ">
+        v-if="$slots.suffix || suffixIcon || showClear || showPassword || isWordLimitVisible || (validateState && needStatusIcon) ">
```
Comment (SimonaliaChen) [inline packages/input/src/input.vue]: 个人认为这么复杂的判断语句 直接写在模板里是不是不太好维护？考虑放在计算属性里吧

Comment (ziyoung) [reply packages/input/src/input.vue]: 有 bug 的。slot 不是响应式的。

**Comment by lvjiaxuan** [comment]
在设置了size的el-form里的el-input，使用show-word-count 有样式问题？？？？？？？

**Pull request merged**